### PR TITLE
Docs: Add dask-gateway and Coiled to list of Dask setup tools

### PR DESCRIPTION
**Pull request by koverholt**
- ~[ ] Tests added / passed~
- ~[ ] Passes `black dask` / `flake8 dask` / `isort dask`~

Per discussion in https://github.com/dask/community/issues/166, this PR:

- Adds `dask-gateway` to list of Dask setup tools
- Adds a `Managed / hosted Dask clusters` section to Dask setup page
- Adds `Coiled` to list of Dask setup tools under `Managed / hosted Dask clusters` section

Note:

Initially I added the `Managed / hosted Dask clusters` section under the `Distributed computing` section, but the bold formatting and spacing looked a bit odd compared to the rest of the list:

Screen Shot 2021-06-16 at 1 09 24 PM

so I moved it to its own top-level section as in:

Screen Shot 2021-06-16 at 1 24 29 PM

### CODE REVIEW

**Comment by jsignell** [comment]
Sure - thanks for the ping! I'll just push to this branch if that's ok with you @koverholt

File: docs/source/setup.rst [hunk 1]
```diff
@@ -77,6 +79,9 @@ how to set up Dask on a variety of local and distributed hardware.
     - :doc:`Cloud <setup/cloud>` for current recommendations on how to
       deploy Dask and Jupyter on common cloud providers like Amazon, Google, or
       Microsoft Azure.
+- Hosted / managed Dask clusters:
+    - `Coiled <https://coiled.io/>`_ handles the creation and management of
+      Dask clusters on cloud computing environments (AWS, Azure, and GCP).
```
Comment (jsignell) [inline docs/source/setup.rst:84]:
```suggestion
    - `Coiled <https://coiled.io/>`_ handles the creation and management of
      Dask clusters on cloud computing environments (AWS, Azure, and GCP).
    - `Saturn Cloud <https://saturncloud.io/>`_ lets users create
      Dask clusters in a hosted platform or within their own AWS accounts.
```

**Comment by jsignell** [comment]
Actually I put the Saturn Cloud bit as a comment, so once that is in I am good.

**Comment by koverholt** [comment]
Thanks @ncclementi and @jsignell, appreciate the updates! I added the description that @jsignell provided and a note about the list being in alphabetical order.

**Comment by ncclementi** [comment]
Thank you @koverholt and @jsignell for your work. This looks ready to go now :tada:

@jrbourbeau would you do the honors and merge this in when CI finishes?

**Comment by jcrist** [comment]
Thanks y'all!

**Pull request merged**